### PR TITLE
4142 limited consent

### PR DESCRIPTION
**Pull request by hopelam**
## Description
https://app.zenhub.com/workspace/o/department-of-veterans-affairs/vets.gov-team/issues/12552

Move limited consent to aggregate level; add text box for description if limited consent checkbox is checked; content of text box is sent to api

## Testing done

local developer testing done

## Screenshots
nothing expanded
all_expanded

## Acceptance criteria
- [ ]  Limited Consent check box is presented at aggregate level and not for every individual provider that is entered
- [ ] The text for the Limited Consent checkbox should be updated to state "I want to limit my consent for the VA to retrieve only specific information from my private medical provider(s)"
- [ ]  Upon the user selecting the Limited Consent check box, then a free text field is presented in the UI 
- [ ] The free text field should have display the following brief instructional text:  "Describe the limitation below (Treatment dates, Disability type, etc.) (*Required)"
- [ ] allow up to 83 characters of free text to be entered in the UI because that is the character limit allowed within the pdf for this field
- [ ] if Limited Consent check box is unchecked, then free text field will remain hidden in UI
- [ ] whatever is entered in the free text field should appear in field 12 of the 4142a pdf that is generated
- [ ] "what does this mean" text should be updated to: "If you choose to limit consent, your doctor will abide by the limitation you specify.  Limiting consent could add to the time it takes to get your private medical records."

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by robfelder — CHANGES_REQUESTED** [review]
Overall everything looks great, just a few minor things here and there. I noticed that the payload that's being sent is sending like this:
```
"form4142":{  
      "limitedConsent":"[object Object] [object Object]",
      "providerFacility":[  
        {  
          "providerFacilityName":"Another Provider",
          "treatmentDateRange":[  
            {  
              "from":"2010-01-03",
              "to":"2012-05-08"
            }
          ],
          "providerFacilityAddress":{  
            "street":"1234 test rd",
            "city":"Testville",
            "postalCode":"12345",
            "country":"USA",
            "state":"AZ"
          }
        },
        {  
          "providerFacilityName":"Another Provider",
          "treatmentDateRange":[  
            {  
              "from":"2003-01-05",
              "to":"2005-10-04"
            }
          ],
          "providerFacilityAddress":{  
            "street":"1234 test rd",
            "city":"Testville",
            "postalCode":"12345",
            "country":"USA",
            "state":"AZ"
          }
        }
      ]
    }
```
Where the limited consent is just [object object], is that intended behavior?
Also, noticed there aren't any unit tests. Were they not needed? At least to validate the payload?

I'm also getting an error submitting the application. It works fine against the `bah-sprint3` branch, but not against the `4141-limited-consent` branch:

screen shot 2018-09-21 at 11 45 15 am
screen shot 2018-09-21 at 11 45 50 am

File: src/applications/disability-benefits/4142/helpers.jsx [hunk 1]
```diff
@@ -474,18 +466,18 @@ export const stateNames = [
  * Converts the treatment date range into an array of objects from just an object
  * @param {object} treatmentDateRange object containing from/to date range
  */
-const transformDateRange = (treatmentDateRange) => {
+const transformDateRange = treatmentDateRange => {
```
Comment (igreulich) [inline src/applications/disability-benefits/4142/helpers.jsx]: This is probably better written utilizing the es6 implicit `return`ing.

`const transformDateRange = treatmentDateRange => ([treatmentDateRange]);`

File: src/applications/disability-benefits/526EZ/helpers.jsx [hunk 2]
```diff
@@ -320,25 +324,26 @@ export const evidenceTypeHelp = (
   </AdditionalInfo>
 );
 
-/**
```
[the remark this answers is not in the document]
Comment (igreulich) [reply src/applications/disability-benefits/526EZ/helpers.jsx]: There is no need, this stopped being a function.

**Review by igreulich — CHANGES_REQUESTED** [review]
I just have the one tweak.

File: src/applications/disability-benefits/4142/helpers.jsx [hunk 3]
```diff
@@ -66,16 +66,6 @@ export const recordHelp = (
   </AdditionalInfo>
 );
 
-export const limitedConsentDescription = (
```
[the remark this answers is not in the document]
Comment (imeecuison) [reply src/applications/disability-benefits/4142/helpers.jsx]: @robfelder: I have moved everything we are using out of 4142 directory and deleted the 4142 directory. This warning should no longer be there. Thank you!

**Comment by imeecuison** [comment]
@robfelder , @igreulich , & @JBarbour2 : I have made the requested changes, please review again. Thank you!

**Comment by imeecuison** [comment]
Just realized something... Unit tests still need to be written. Should I write them before you review again? @robfelder @igreulich

File: src/applications/disability-benefits/526EZ/helpers.jsx [hunk 4]
```diff
@@ -320,25 +324,26 @@ export const evidenceTypeHelp = (
...
- * Grabbing the disbility string and returning a display content. Similar to getEvidenceTypesDescription. Maybe refactor?
- * @param {object} form Form data
- * @param {*} index identifier for which disability/contention is being evaluated
- */
-export const getlimitedConsentTitle = (form, index) => {
-  const { name } = form.disabilities[index];
-  return limitedConsentTitle(getDisabilityName(name));
-};
+export const limitedConsentTextTitle = (
+  <p>Describe the limitation here. (Treatment dates, Disability type, etc.)</p>
```
[the remark this answers is not in the document]
Comment (igreulich) [reply src/applications/disability-benefits/526EZ/helpers.jsx]: @robfelder Are you referring to the lack of (*Required), or the addition of the . after the word below?

**Review by igreulich — APPROVED** [review]

**Comment by igreulich** [comment]
@imeecuison I gave you some merge conflicts. I also might fix them, in #8487

File: src/applications/disability-benefits/526EZ/helpers.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (robfelder) [reply src/applications/disability-benefits/526EZ/helpers.jsx]: No, it says `Describe the limitation here`, but the acceptance criteria says `Describe the limitation below`

[the remark this answers is not in the document]
Comment (igreulich) [reply src/applications/disability-benefits/526EZ/helpers.jsx]: I totally missed that one.

@imeecuison are you working on these fixes?

[the remark this answers is not in the document]
Comment (imeecuison) [reply src/applications/disability-benefits/526EZ/helpers.jsx]: This has been changed to Describe the limitation below.

**Review by robfelder — APPROVED** [review]

**Pull request merged**